data/org.freedesktop.portal.GlobalShortcuts.xml

```diff
@@ -0,0 +1,195 @@
+        @request_handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Shows a dialog with all shortcuts to configure and offers them back to the app when done.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>shortcuts a(sa{sv})</term>
+            <listitem><para>
+              The different shortcut ids that have been registered. They are mapped to a list of pairs including:
+                - description: User-readable text describing what the shortcut does
+                - trigger: User-readable text describing how to trigger the shortcut
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="BindShortcuts">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="as" name="shortcuts" direction="in"/>
+      <arg type="s" name="parent_window" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="request_handle" direction="out"/>
+
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In3" value="QVariantMap"/>
+    </method>
+
+    <!--
+        Activated:
+        Notifies about a shortcut becoming active.
+
+        @session_handle: Session that requested the shortcut
+        @shortcut_id: the application-provided ID for the notification
+        @timestamp: When the event happened in µs
+        @options: unused
+    -->
+    <signal name="Activated">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="shortcut_id" direction="in"/>
+      <arg type="t" name="timestamp" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In4" value="QVariantMap"/>
+    </signal>
+
+    <!--
+        Deactivated:
+        Notifies that a shortcut is not active anymore.
+
+        @session_handle: Session that requested the shortcut
+        @shortcut_id: the application-provided ID for the notification
+        @timestamp: When the event happened in µs
+        @options: unused
+    -->
+    <signal name="Deactivated">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="shortcut_id" direction="in"/>
+      <arg type="t" name="timestamp" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <annotation name="org.qtproject.QtDBus.QtTypeName.In4" value="QVariantMap"/>
+    </signal>
+
+    <!--
+        ShortcutsChanged:
+        @session_handle: Session that requested the shortcut
+        @shorcuts: The different shortcut ids that have been registered. They are mapped to a list of pairs including:
+          - description: User-readable text describing what the shortcut does
+          - trigger: User-readable text describing how to trigger the shortcut
```

Comment:
I tend to agree.

Reply:
I'm not sure what you mean, what would that look like?

Reply:
The idea would be to have a <vararg> list in some generic top level like documentation section, and everywhere where the same format is expected to be passed, the documentation will say "See ... for details about the shortcut vararg." or something, so that when we e.g. add a new entry to the dictionary, it'll only be added to one place.

Reply:
Ah right, makes sense. I've put it on the class definition and pointed there. Not sure if there's a better way of doing this.